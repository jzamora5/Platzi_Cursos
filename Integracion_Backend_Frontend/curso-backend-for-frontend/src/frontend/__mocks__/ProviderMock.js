import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import initialState from '../initialState';
import reducer from '../reducers';

const store = createStore(
  reducer,
  initialState,
  compose(applyMiddleware(thunk))
);

const history = createBrowserHistory();

const ProviderMock = (props) => (
  <Provider store={store}>
    <Router history={history}>{props.children}</Router>
  </Provider>
);

module.exports = ProviderMock;
