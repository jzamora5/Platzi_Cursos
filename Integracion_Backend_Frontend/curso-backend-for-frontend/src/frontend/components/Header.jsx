import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { logoutRequest } from "../actions";
import gravatar from '../utils/gravatar';
import '../assets/styles/components/Header.scss';
import logo from '../assets/static/logo-platzi-video-BW2.png';
import userIcon from '../assets/static/user-icon.png';

const Header = props => {
  const { user, isLogin, isRegister } = props;
  const hasUser = Object.keys(user).length > 0;
  const HeaderClass = classNames('header', {
    isLogin,
    isRegister,
  });
  const handleLogout = () => {
    document.cookie = "email=";
    document.cookie = "name=";
    document.cookie = "id=";
    document.cookie = "token="
    props.logoutRequest({});
    window.location.href = '/login'
  };
  return (
    <header className={HeaderClass}>
      <Link to="/">
        <img className="header__img" src={logo} alt="Platzi Video" />
      </Link>
      <div className="header__menu">
        <div className="header__menu--profile">
          {hasUser ? <img src={gravatar(user.email)} alt={user.email} /> : <img src={userIcon} alt="Usuario" />}
          <p>Perfil</p>
        </div>
        <ul>
          {hasUser ? <li><a href="/">{user.name}</a></li> : null}
          {hasUser ? <li><a href="#logout" onClick={handleLogout}>Cerrar Sesión</a></li> : <li><Link to="login">Iniciar Sesión</Link></li>}
        </ul>
      </div>
    </header>
  );
}

const mapStateToProps = state => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = {
  logoutRequest,
};

Header.propTypes = {
  user: PropTypes.object,
  isLogin: PropTypes.bool,
  isRegister: PropTypes.bool,
  logoutRequest: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);