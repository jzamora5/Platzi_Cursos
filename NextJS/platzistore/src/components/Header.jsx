import React, { useContext } from 'react';
import Image from 'next/image';
import Menu from '@components/Menu';
import MyOrder from '@containers/MyOrder';
import menu from '@icons/icon_menu.svg';
import logo from '@logos/logo_yard_sale.svg';
import AppContext from '@context/AppContext';
import shoppingCart from '@icons/icon_shopping_cart.svg';
import styles from '@styles/Header.module.scss';

const Header = () => {
	const { state, toggleOrder, toggleMenu } = useContext(AppContext);

	return (
		<>
			<nav className={styles.Nav}>
				<Image src={menu} alt="menu" className={styles.menu} />
				<div className={styles['navbar-left']}>
					<Image src={logo} alt="logo" className={styles['nav-logo']} />
					<ul>
						<li>
							<a href="/">All</a>
						</li>
						<li>
							<a href="/">Clothes</a>
						</li>
						<li>
							<a href="/">Electronics</a>
						</li>
						<li>
							<a href="/">Furnitures</a>
						</li>
						<li>
							<a href="/">Toys</a>
						</li>
						<li>
							<a href="/">Others</a>
						</li>
					</ul>
				</div>
				<div className={styles['navbar-right']}>
					<ul>
						<li className={styles['more-clickable-area'], styles['navbar-email'], styles.pointer} onClick={() => toggleMenu()}>
							platzi@example.com
						</li>
						<li
							// className={styles.['navbar-shopping-cart']}
							onClick={() => toggleOrder()}
						>
							<Image className={styles['more-clickable-area'], styles.pointer} src={shoppingCart} alt="shopping cart" />
							{state.cart.length > 0 ? <div>{state.cart.length}</div> : null}
						</li>
					</ul>
				</div>
				{state.menuIsOpen && <Menu />}
			</nav>
			{state.orderIsOpen && <MyOrder />}
		</>
	);
}

export default Header;
