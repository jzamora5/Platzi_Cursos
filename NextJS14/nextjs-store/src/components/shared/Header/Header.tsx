import Link from "next/link";

export const Header = () => {
  return (
    <header>
      <nav>
        <ul>
          <Link href="/">
            <li>Home</li>
          </Link>
          <Link href="/store">
            <li>Store</li>
          </Link>
        </ul>
      </nav>
    </header>
  );
};
