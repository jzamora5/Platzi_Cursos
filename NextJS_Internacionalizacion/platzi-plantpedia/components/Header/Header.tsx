import { PropsWithChildren } from 'react'
import Link, { LinkProps } from 'next/link'
import { NavBar } from '@ui/NavBar'
import { Button } from '@ui/Button'

import { PreviewModeBanner } from './PreviewModeBanner'

export function Header() {
  return (
    <>
      <PreviewModeBanner />
      <div className="mx-auto" style={{ maxWidth: '98%' }}>
        <NavBar title="ðŸŒ¿ Plantpedia">
          <div>
            <NavLink href="/top-stories">Top Stories</NavLink>
          </div>
        </NavBar>
      </div>
    </>
  )
}

function NavLink({ children, ...linkProps }: PropsWithChildren<LinkProps>) {
  return (
    <Link {...linkProps} passHref>
      <Button color="inherit" variant="text" component="a">
        {children}
      </Button>
    </Link>
  )
}
