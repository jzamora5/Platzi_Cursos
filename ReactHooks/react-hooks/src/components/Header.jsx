import { useState, useContext } from "react";
import ThemeContext from "../context/ThemeContext";

const Header = () => {
  const [darkMode, setDarkMode] = useState(false);
  const color = useContext(ThemeContext);

  const handleClick = () => {
    setDarkMode(!darkMode);
  };

  return (
    <div className="Header">
      <h1 style={{ color }}>ReactHooks</h1>
      <button type="button" onClick={handleClick}>
        {darkMode ? "Dark Mode" : "Light Mode"}
      </button>
      <button type="button" onClick={() => setDarkMode(!darkMode)}>
        {darkMode ? "Dark Mode 2" : "Light Mode 2"}
      </button>
    </div>
  );
};

export default Header;
