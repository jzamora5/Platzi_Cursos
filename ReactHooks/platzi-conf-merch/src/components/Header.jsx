import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import AppContext from '../context/AppContext';
import '../styles/components/Header.css';

const Header = () => {
  const { state } = useContext(AppContext);
  const { cart } = state;

  return (
    <div className="Header">
      <Link to="/">
        <h1 className="Header-title">PlatziConf Merch</h1>
      </Link>
      <div className="Header-checkout">
        <Link to="/checkout">
          <i className="fas fa-shopping-basket"></i>
        </Link>
        {cart.length > 0 && <div className="Header-alert">{cart.length}</div>}
      </div>
    </div>
  );
};

export default Header;
