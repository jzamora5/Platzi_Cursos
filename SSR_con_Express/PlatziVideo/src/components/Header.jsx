import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import gravatar from '../utils/gravatar';
import { logoutRequest } from '../actions';
import '../assets/styles/components/Header.scss';
import logo from '../assets/static/logo-platzi-video-BW2.png';
import userIcon from '../assets/static/user-icon.png';

const Header = props => {
  const { user, isLogin, isRegister } = props;
  const hasUser = Object.keys(user).length > 0;

  const handleLogout = () => {
    props.logoutRequest({});
  }
  const headerClass = classNames('header', {
    isLogin,
    isRegister,
  });
  return (
    <header className={headerClass}>
      <Link to="/">
        <img className="header__img" src={logo} alt="Platzi Video" />
      </Link>
      <div className="header__menu">
        <div className="header__menu--profile">
          {hasUser ?
            <img src={gravatar(user.email)} alt={user.email} /> :
            <img src={userIcon} alt="" />
          }
          <p>Perfil</p>
        </div>
        <ul>
          {hasUser ?
            <li><a href="/">{user.name}</a></li> 
            : null
          }
          {hasUser ? 
            <li><a href="#logout" onClick={handleLogout}>Cerrar Sesión</a></li>
            :
            <li>
              <Link to="/login">
                Iniciar sesión
              </Link>
            </li>
          }
        </ul>
      </div>
    </header>
  );
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

const mapDispatchToProps = {
  logoutRequest,
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);