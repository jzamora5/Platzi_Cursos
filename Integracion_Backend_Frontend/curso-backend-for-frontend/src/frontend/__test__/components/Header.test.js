import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';
import ProviderMock from '../../__mocks__/ProviderMock';

describe('<Header /> ', () => {
  test('Header logo image', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.find('.header__img')).toHaveLength(1);
  });

  test('Header snapshot', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
