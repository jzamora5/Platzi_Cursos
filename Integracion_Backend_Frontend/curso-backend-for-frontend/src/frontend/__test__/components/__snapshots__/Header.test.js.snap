// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />  Header snapshot 1`] = `
<header
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Platzi Video"
      className="header__img"
      src="test-file-stub"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt="Usuario"
        src="test-file-stub"
      />
      <p>
        Perfil
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesi√≥n
        </a>
      </li>
    </ul>
  </div>
</header>
`;
